atlasdb-api/src/main/java/com/palantir/atlasdb/metrics/MetricPublicationArbiter.java

```diff
@@ -69,4 +80,41 @@ private static boolean safeShouldPublish(MetricName metricName, MetricPublicatio
             return true;
         }
     }
+
+    /**
+     * Most users will want to define filters without having to think about deduplicating
+     * filters across calls (similar to how MetricsRegistry methods are all registerOrGet)
+     * so in order to keep the existing ability to define filters with lambdas,
+     * define a default method on {@link MetricPublicationFilter} with a reasonable deduplicator
+     * and then wrap it here.
+     */
+    @VisibleForTesting
+    static class DeduplicatingFilterHolder {
```

Comment:
Is there any reason why this does not use `@Value.Immutable`? As far as I can tell, the equality check is very similar to the Immutables one, if not exactly equivalent.

Reply:
Not that I'm aware of (other than this being what @dxiao had written)

Reply:
no reason, it was just faster to write.